shared/fs/filepreview/bare-preview.native.js

```diff
@@ -36,65 +36,87 @@ type ConnectedBarePreviewProps = {
   onBack: () => void,
 }
 
-const BarePreview = (props: ConnectedBarePreviewProps) => (
-  <Box style={stylesContainer}>
-    <Box style={stylesHeader}>
-      <ClickableBox onClick={props.onBack} style={stylesCloseBox}>
-        <Text type="Body" style={stylesText}>
-          Close
-        </Text>
-      </ClickableBox>
-    </Box>
-    <Box style={stylesContentContainer}>
-      <View path={props.path} routePath={props.routePath} />
-    </Box>
-    <Box style={stylesFooter}>
-      <PathItemAction path={props.path} actionIconStyle={stylesPathItemActionIcon} />
-    </Box>
-  </Box>
-)
+type State = {
+  loading: boolean,
+}
+
+class BarePreview extends React.PureComponent<ConnectedBarePreviewProps, State> {
+  state = {
+    loading: false,
+  }
+  _onLoadingStateChange = (loading: boolean) => this.setState({loading})
 
-const stylesPathItemActionIcon = {
-  color: globalColors.white,
+  render() {
+    return (
+      <Box style={styles.container}>
+        <Box style={styles.header}>
+          <ClickableBox onClick={this.props.onBack} style={styles.closeBox}>
+            <Text type="Body" style={styles.text}>
+              Close
+            </Text>
+          </ClickableBox>
+        </Box>
+        <Box style={styles.contentContainer}>
+          <View
+            path={this.props.path}
+            routePath={this.props.routePath}
+            onLoadingStateChange={this._onLoadingStateChange}
+          />
+        </Box>
+        <Box style={styles.footer}>
+          <PathItemAction path={this.props.path} actionIconWhite={true} />
+        </Box>
+        {this.state.loading && <ProgressIndicator style={styles.loading} white={true} />}
+      </Box>
+    )
+  }
 }
 
-const stylesContainer = platformStyles({
-  common: {
-    ...globalStyles.flexBoxColumn,
-    ...globalStyles.flexGrow,
-    backgroundColor: globalColors.black,
+const styles = Styles.styleSheetCreate({
+  container: Styles.platformStyles({
+    common: {
+      ...Styles.globalStyles.flexBoxColumn,
+      ...Styles.globalStyles.flexGrow,
+      backgroundColor: Styles.globalColors.black,
+    },
+    isIOS: {
+      marginTop: -20, // top status bar
```

Comment:
i know this is old but you shouldn't need this. you can define the route leaf tags to not be under the status bar

Reply:
did you try underStatusbar

Reply:
Just tried; it worked!